lib/Interpreter/CppInterOp.cpp

```diff
@@ -1657,6 +1657,21 @@ namespace Cpp {
     return QT.getCanonicalType().getAsOpaquePtr();
   }
 
+  bool IsRestrictType(TCppType_t type) {
+    QualType QT = QualType::getFromOpaquePtr(type);
+    return QT.isRestrictQualified();
+  }
+
+  TCppType_t GetNonRestrictType(TCppType_t type) {
+    QualType QT = QualType::getFromOpaquePtr(type);
+    if (QT.isRestrictQualified()) {
+      QualType NonRestrictType(QT);
+      NonRestrictType.removeLocalRestrict();
+      return NonRestrictType.getAsOpaquePtr();
+    }
+    return type;
```

Comment:
Can we make codecov happy here?

Reply:
I think we need a generic interface for the cvr types. Something like:

bool HasTypeQualification(Quals::Restrict & Quals::Const & Quals::Volatile) { ... }

The getter should be taking the same...

Reply:
Yes. That can be done.
```cpp
enum Qual { Restrict = 0b01, Const = 0b010, Volatile = 0b0100 };

bool HasTypeQualifier(TCppType_t, enum Qual) {
    ...
}

TCppType RemoveTypeQualifier(TCppType_t, enum Qual) {
    ...
}
```
The `enum` can be simply `xor`ed to combine different qualifiers.

@vgvassilev, Is this approach good?

Reply:
Yes, makes sense. The enum should be const, volatile, restrict to follow the CVR terminology.